js/client-chat.js

```diff
@@ -635,35 +635,45 @@
 			case 'showjoins':
 				var showjoins = Dex.prefs('showjoins') || {};
 				var serverShowjoins = showjoins[Config.server.id] || {};
-				if (target) {
-					var room = toID(target);
-					if (serverShowjoins['global']) {
-						delete serverShowjoins[room];
-					} else {
-						serverShowjoins[room] = 1;
-					}
-					this.add('Join/leave messages on room ' + room + ': ALWAYS ON');
-				} else {
-					serverShowjoins = {global: 1};
-					this.add('Join/leave messages: ALWAYS ON');
+				switch (target) {
+				case 'global':
+					serverShowjoins['global'] = 1;
+					this.add('Join/leave messages' + ': ALWAYS ON');
+					break;
+				case 'chatrooms':
+					serverShowjoins['chatrooms'] = 1;
+					this.add('Join/leave messages in chatrooms' + ': ALWAYS ON');
+					break;
+				case 'battles':
+					serverShowjoins['battles'] = 1;
+					this.add('Join/leave messages in battle' + ': ALWAYS ON');
+					break;
+				default:
+					serverShowjoins[this.id] = 1;
```

Comment:
It might be excessive to store these settings forever in battles that auto-expire after 10 minutes.